### DISCUSSION TITLE
Routing with regex

### DISCUSSION THREAD

**Discussion by jneto81** [activej/activej#211] [General]
Is it possible to use Regular expressions for routing the requests ?

The RoutingServlet is final, so i can't see how to make this possible.

**Comment by eduard-vasinskyi** [comment]
Hi, @jneto81.

RoutingServlet is optimized for performance, so it only allows simple routing.

However, what you could do is redirect the request manually by checking the regex path part of your request and then serve a request with the right servlet:
```java
RoutingServlet.create()
    .map(GET, "/user/:regex", request -> {
      String regex = request.getPathParameter("regex");
      if (pattern.matcher(regex).matches()){
         return matchServlet.serve(request);
      } else {
         return noMatchServlet.serve(request);
      }
})
```
